cg/cli/generate/report/base.py

```diff
@@ -61,22 +57,25 @@ def generate_delivery_report(
         click.echo(delivery_report_html)
         return
 
-    delivery_report_file: TextIO = report_api.create_delivery_report_file(
-        case_id,
+    version: Version = report_api.housekeeper_api.version(bundle=case_id, date=analysis_date)
+    delivery_report: Optional[str] = report_api.get_delivery_report_from_hk(
+        case_id=case_id, version=version
+    )
+    if delivery_report:
```

Comment:
Is this the usage we want?
If we are calling from the command the command `cg generate delivery-report` wouldn't we want to generate a delivery report even if it's already in housekeeper?  It might be that we want to overwrite a previous one that is already storage (i.e. if the first one had some error). I understand that this is safer, but this would entail manually remove previous delivery reports from housekeeper and then generate the report.  I just want to make sure that is what we want.

Reply:
I will try to address your comments and clarify the concept for myself 😆 

In Scout, each analysis version is associated with a specific date when it's stored. As a result, we have a unique delivery report for each analysis version. In order to maintain traceability, when we upload a new version of an existing analysis, certain results may be overwritten, but we still keep track of the delivery report. This logic is implemented on the Scout side.

<img width="155" alt="Screenshot 2023-07-17 at 12 21 28" src="https://github.com/Clinical-Genomics/cg/assets/26309194/4a9ab4a2-6e8e-415b-86f1-d3828e508ba1">

I haven't made any changes to the existing logic because we were already checking if a delivery report was present in HK before. However, I have made it version-dependent, which is precisely what we want as we aim to have one delivery report per version.

Regarding your question:
`Wouldn't we want to generate a delivery report even if it's already in housekeeper?`
- Yes, but only if it belongs to a different version of the analysis. What we want to avoid is uploading or storing incomplete or erroneous delivery reports (which I don't think is the intention of having it versioned).